app/ts/components/App.tsx

```diff
@@ -33,6 +34,7 @@ export function App() {
 	const isWindowEthereum = useSignal<boolean>(true)
 	const petalLockDeployed = useSignal<boolean | undefined>(undefined)
 	const account = useSignal<AccountAddress | undefined>(undefined)
+	const chainId = useSignal<number | undefined>(undefined)
```

Comment:
This information can be memoized in the account object, so why do we store it duplicated here?  Is it to try to avoid `await` everywhere?

Reply:
what do you mean by account object? Do you mean to make object that has address and chain id together, instead of having them separately?

Reply:
Yeah, I guess right now the chainId is not memoized by the account object.  Probably not worth it for this small app, so feel free to ignore.

Reply:
Why do you feel it should memoized to it? Its a separate parameter and its handled separately too

Reply:
Ah, I see my mistake.  I thought `account` was actually an `Account` object that had more to it, but it is just an address.  Maybe rename `account` to `accountAddress` instead so future readers don't get similarly confused.  😬